cdap-app-fabric/src/main/java/io/cdap/cdap/security/impersonation/DefaultUGIProvider.java

```diff
@@ -172,4 +200,52 @@ private File localizeKeytab(Location keytabLocation) throws IOException {
 
     return localKeytabFile.toFile();
   }
+
+  private Map<String, String> getRuntimeImpersonationProperties(NamespacedEntityId programId) {
+      if (programId instanceof ProgramRunId) {
```

Comment:
Check for the negative case and bail early. This makes the code cleaner and less indented, hence easier to read.

```
if (!(programId instanceOf ProgramRunId)) {
  return Collections.emptyMap();
}
// The rest of the logic.
```